src/github.com/matrix-org/dendrite/cmd/mediaapi-integration-tests/main.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/matrix-org/dendrite/common/test"
+)
+
+var (
+	// How long to wait for the server to write the expected output messages.
+	// This needs to be high enough to account for the time it takes to create
+	// the postgres database tables which can take a while on travis.
+	timeoutString = test.Defaulting(os.Getenv("TIMEOUT"), "10s")
+	// The name of maintenance database to connect to in order to create the test database.
+	postgresDatabase = test.Defaulting(os.Getenv("POSTGRES_DATABASE"), "postgres")
+	// The name of the test database to create.
+	testDatabaseName = test.Defaulting(os.Getenv("DATABASE_NAME"), "mediaapi_test")
+	// Postgres docker container name (for running psql)
+	postgresContainerName = os.Getenv("POSTGRES_CONTAINER")
+)
+
+var thumbnailPregenerationConfig = (`
+thumbnail_sizes:
+- width: 32
+  height: 32
+  method: crop
+- width: 96
+  height: 96
+  method: crop
+- width: 320
+  height: 240
+  method: scale
+- width: 640
+  height: 480
+  method: scale
+- width: 800
+  height: 600
+  method: scale
+`)
+
+const serverType = "media-api"
+
+var testDatabaseTemplate = "dbname=%s sslmode=disable binary_parameters=yes"
+
+var timeout time.Duration
+
+func startMediaAPI(suffix string, dynamicThumbnails bool) (*exec.Cmd, chan error, string) {
+	dir, err := ioutil.TempDir("", serverType+"-server-test"+suffix)
+	if err != nil {
+		panic(err)
+	}
+
+	configFilename := serverType + "-server-test-config" + suffix + ".yaml"
+	configFileContents := makeConfig(suffix, dir, dynamicThumbnails)
+
+	serverArgs := []string{
+		"--config", configFilename,
+		"--listen", "localhost:1777" + suffix,
+	}
+
+	databases := []string{
+		testDatabaseName + suffix,
+	}
+
+	cmd, cmdChan := test.StartServer(
+		serverType,
+		serverArgs,
+		suffix,
+		configFilename,
+		configFileContents,
+		postgresDatabase,
+		postgresContainerName,
+		databases,
+	)
+	return cmd, cmdChan, dir
+}
+
+func makeConfig(suffix, basePath string, dynamicThumbnails bool) string {
+	return fmt.Sprintf(
+		`
+server_name: "%s"
+base_path: %s
+max_file_size_bytes: %s
+database: "%s"
+dynamic_thumbnails: %s
+%s`,
+		"localhost:1777"+suffix,
```

Comment:
The name of the server is the "public" address for federation traffic, so will probably be different to the address that the media-api-server is listening on. The media-api-server won't be receiving federation traffic directly since it will need to go through a proxy to add the `/api/` prefix to the path.